test/rpc_test.py

```diff
@@ -72,6 +72,36 @@ def test_retry_rpc_limited(self):
         fetch_results = [socket.timeout, socket.timeout, socket.timeout]
         self.assertRaises(luigi.rpc.RPCError, self.get_work, fetch_results)
 
+    @mock.patch('luigi.rpc.logger')
+    def test_log_rpc_retries_enabled(self, mock_logger):
+        """
+        Tests that each retry of an RPC method is logged
+        """
+
+        fetch_results = [socket.timeout, socket.timeout, '{"response":{}}']
+        self.get_work(fetch_results)
+        self.assertEqual([
+            mock.call.warning('Failed connecting to remote scheduler %r', 'http://zorg.com', exc_info=True),
+            mock.call.info('Retrying attempt 2 of 3 (max)'),
+            mock.call.warning('Failed connecting to remote scheduler %r', 'http://zorg.com', exc_info=True),
+            mock.call.info('Retrying attempt 3 of 3 (max)'),
+        ], mock_logger.mock_calls)
+
+    @with_config({'core': {'rpc-log-retries': 'false'}})
+    @mock.patch('luigi.rpc.logger')
+    def test_log_rpc_retries_disabled(self, mock_logger):
+        """
+        Tests that retries of an RPC method are not logged
+        """
+
+        fetch_results = [socket.timeout, socket.timeout, socket.gaierror]
+        try:
+            self.get_work(fetch_results)
+            self.fail("get_work should have thrown RPCError")
+        except luigi.rpc.RPCError as e:
```

Comment:
Can't you use self.assertRaises?

Reply:
This was done to assert about e.sub_exception.